packages/mermaid/src/diagrams/class/parser/classDiagram.jison

```diff
@@ -274,18 +274,21 @@ statement
     ;
 
 namespaceStatement
-    : namespaceIdentifier STRUCT_START classStatements STRUCT_STOP          { yy.addClassesToNamespace($1, $3); }
-    | namespaceIdentifier STRUCT_START NEWLINE classStatements STRUCT_STOP  { yy.addClassesToNamespace($1, $4); }
+    : namespaceIdentifier STRUCT_START classStatements STRUCT_STOP          { yy.addClassesToNamespace($1, $3[0], $3[1]); }
+    | namespaceIdentifier STRUCT_START NEWLINE classStatements STRUCT_STOP  { yy.addClassesToNamespace($1, $4[0], $4[1]); }
     ;
 
 namespaceIdentifier
     : NAMESPACE namespaceName { $$=$2; yy.addNamespace($2); }
     ;
 
 classStatements
-    : classStatement                            {$$=[$1]}
-    | classStatement NEWLINE                    {$$=[$1]}
-    | classStatement NEWLINE classStatements    {$3.unshift($1); $$=$3}
+    : classStatement                            {$$=[[$1], []]}
```

Comment:
Can you help me understand this change? What do the extra array definitions do?

Reply:
Thank you for the review!

In short, this is to pass not only class node identifiers but also note node identifiers to `yy.addClassesToNamespace()`.

`classStatements` is an array of identifiers for class nodes within the namespace.

Originally, only regular class nodes were stored in this array. These were passed to `yy.addClassesToNamespace()` within `namespaceStatement`, linking them to the namespace.

My change extends the format of `classStatements` from `[classIdentifiers]` to `[[classIdentifiers], [noteIdentifiers]]`. This allows note node identifiers to be passed to `yy.addClassesToNamespace()` and linked to the namespace.

Reply:
Of course, for better extensibility and readability, we could use an object (associative array) instead of an array, like this:
```
classStatement NEWLINE                   {$$={classNames: [$1], noteNames: []}
...
```
However, I thought it might be overkill to do so just for the two types: class nodes and note nodes.

Reply:
Which do you think is better? I don’t have strong feelings about it, so I can rewrite it quickly.

Reply:
The way you have done it is fine, it just took your explanation to help me see it :)